workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/AlertOverlayDialogFactory.kt

```diff
@@ -53,29 +52,26 @@ public open class AlertOverlayDialogFactory : OverlayDialogFactory<AlertOverlay>
           setButton(button.toId(), " ") { _, _ -> }
         }
       }
-  }
-
-  open override fun updateDialog(
-    dialog: Dialog,
-    rendering: AlertOverlay,
-    environment: ViewEnvironment
-  ) {
-    (dialog as AlertDialog).apply {
```

Comment:
See how we're able to get rid of this cast? That's a big benefit of this approach.